src/main/java/com/laytonsmith/core/AliasCore.java

```diff
@@ -344,121 +326,218 @@ public final void reload(MCPlayer player, String[] settings, boolean firstLoad)
 				}
 			}
 
-			if(!Prefs.isInitialized()) {
-				Prefs.init(prefFile);
+			// Get the default files (main.ms, aliases.msa, auto_include.ms)
+			try {
+				String main = file_get_contents(mainFile.getAbsolutePath());
+				localPackages.appendMS(main, mainFile);
+			} catch (IOException e) {
+				Static.getLogger().log(Level.WARNING, "Could not read main file");
 			}
 
-			if(options.reloadScripts()) {
-				ProfilePoint unregisteringEvents = parent.profiler.start("Unregistering events", LogLevel.VERBOSE);
-				try {
-					EventUtils.UnregisterAll();
-				} finally {
-					unregisteringEvents.stop();
-				}
-				ProfilePoint runningExtensionHooks = parent.profiler.start("Running event hooks", LogLevel.VERBOSE);
-				try {
-					ExtensionManager.RunHooks();
-				} finally {
-					runningExtensionHooks.stop();
-				}
+			try {
+				String aliasConfigStr = file_get_contents(aliasConfig.getAbsolutePath());
+				localPackages.appendMSA(aliasConfigStr, aliasConfig);
+			} catch (IOException e) {
+				Static.getLogger().log(Level.WARNING, "Could not read aliases file");
+			}
+
+			File autoInclude = new File(env.getEnv(GlobalEnv.class).GetRootFolder(), "auto_include.ms");
+			if(autoInclude.exists()) {
+				localPackages.addAutoInclude(autoInclude);
+			}
 
-				//Clear the include cache, so it re-pulls files.
-				IncludeCache.clearCache();
+			// Now that the default files are added, search the LocalPackages directory
+			localPackages.search(fileLocations.getLocalPackagesDirectory());
+			includeCache.addAutoIncludes(localPackages.getAutoIncludes());
 
-				// Close all channel messenger channels registered by CH.
-				Static.getServer().getMessenger().closeAllChannels();
+			ProfilePoint compilerMS = profiler.start("Compilation of MS files in Local Packages", LogLevel.VERBOSE);
 
-				// Unregister previously registered commands
-				{
-					MCCommandMap map = Static.getServer().getCommandMap();
-					for(String command : registeredCommands) {
-						map.getCommand(command).unregister(map);
+			// Set and analyze auto includes for static analysis.
+			Set<ConfigCompileException> compileExceptions = new HashSet<>();
+			StaticAnalysis.setAndAnalyzeAutoIncludes(
```

Comment:
This is still static, which forms an issue when 'reverting back' (or rather not touching) to the old execution environment. If in that execution environment a call is made to `eval()`, or something is dynamically `include()`d, it will use the newly set auto includes to perform static analysis on. I suppose that `StaticAnalysis.autoIncludesAnalysis` could be moved to the `StaticRuntimeEnv` environment instead (just like the `IncludeCache`), resulting in having one shared `autoIncludesAnalysis` per recompile. This does not affect LangServ for now, as it has no notion of auto includes.